app/assets/javascripts/discourse/app/components/quote-button.js

```diff
@@ -99,31 +96,38 @@ export default Component.extend({
     let firstRange, postId;
     for (let r = 0; r < selection.rangeCount; r++) {
       const range = selection.getRangeAt(r);
-      const $selectionStart = $(range.startContainer);
-      const $ancestor = $(range.commonAncestorContainer);
-
-      if ($selectionStart.closest(".cooked").length === 0) {
+      const selectionStart =
+        range.startContainer.nodeType === Node.ELEMENT_NODE
+          ? range.startContainer
+          : range.startContainer.parentElement;
+      const ancestor =
+        range.commonAncestorContainer.nodeType === Node.ELEMENT_NODE
+          ? range.commonAncestorContainer
+          : range.commonAncestorContainer.parentElement;
+
+      if (!selectionStart.closest(".cooked")) {
         return;
       }
 
       firstRange = firstRange || range;
-      postId = postId || $ancestor.closest(".boxed, .reply").data("post-id");
+      postId = postId || ancestor.closest(".boxed, .reply")?.dataset.postId;
```

Comment:
This will blow up if `ancestor.closest(".boxed, .reply")` is undefined right?

```suggestion
      postId = postId || ancestor.closest(".boxed, .reply")?.dataset?.postId;
```